File: backend/app/schemas/alert.py
"""Response schema for alert endpoints."""

import uuid
from datetime import datetime

from pydantic import BaseModel


class AlertResponse(BaseModel):
    """Schema for alert data in API responses."""

    id: uuid.UUID
    price_watch_id: uuid.UUID
    user_id: uuid.UUID
    price_snapshot_id: uuid.UUID
    alert_type: str
    channel: str
    status: str
    target_price: int
    triggered_price: int
    message: str | None
    sent_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: ios/TravelPrice/TravelPrice/Core/Models/Alert.swift
import Foundation

struct PriceAlert: Codable, Identifiable {
    let id: String
    let priceWatchId: String
    let userId: String
    let priceSnapshotId: String
    let alertType: String
    let channel: String
    let status: String
    let targetPrice: Int
    let triggeredPrice: Int
    let message: String?
    let sentAt: String?
    let createdAt: String

    /// How much below target the triggered price was, in cents.
    var savingsCents: Int {
        targetPrice - triggeredPrice
    }
}
